data-prepper-plugins/peer-forwarder/src/test/java/com/amazon/dataprepper/plugins/processor/peerforwarder/discovery/DnsPeerListProviderTest.java

```diff
@@ -0,0 +1,56 @@
+package com.amazon.dataprepper.plugins.processor.peerforwarder.discovery;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.dns.DnsAddressEndpointGroup;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DnsPeerListProviderTest {
+    private static final String ENDPOINT_1 = "10.1.1.1";
+    private static final String ENDPOINT_2 = "10.1.1.2";
+    private static final List<Endpoint> ENDPOINT_LIST = Arrays.asList(
+            Endpoint.of(ENDPOINT_1),
+            Endpoint.of(ENDPOINT_2)
+    );
+
+    @Mock
+    private DnsAddressEndpointGroup dnsAddressEndpointGroup;
+    @Mock
+    private CompletableFuture completableFuture;
```

Comment:
How about using `CompletableFuture.completedFuture(null)`, according to https://www.baeldung.com/java-completablefuture:

If we already know the result of a computation, we can use the static completedFuture method with an argument that represents a result of this computation. Consequently, the get method of the Future will never block, immediately returning this result instead:

Future<String> completableFuture = 
  CompletableFuture.completedFuture("Hello");

// ...

String result = completableFuture.get();
assertEquals("Hello", result);

Reply:
That's better, thanks